main.js

```diff
@@ -0,0 +1,235 @@
+    var capitalAbbr = abbr.toUpperCase(); 
+    state = states[capitalAbbr]; 
+    fullName = str1 + ', ' + state; 
+    return fullName; 
+
+  } else {
+    fullName = city; 
+    return fullName
+  }; 
+};
+
+var renderCurrentWeather = function () {
+  $('.current-weather').empty();
+
+  for (let i = 0; i < currentWeather.length; i++) {
+    var source = $('#current-weather-template').html();
+    var template = Handlebars.compile(source);
+    var currentWeatherHTML = template(currentWeather[i]);
+    $('.current-weather').append(currentWeatherHTML); 
+
+    var iconSource = $('#current-weather-icon-template').html();
+    var iconTemplate = Handlebars.compile(iconSource);
+    var currentWeatherIconHTML = iconTemplate(currentWeather[i]);
+    $('.current-weather').append(currentWeatherIconHTML); 
+  };
+   changeStyle(); 
+}; 
+
+var addWeatherForecast = function (data) {
+  var forecastData = data.list; 
+  dailyData = compileFetchData(forecastData, 'dt_txt');
+  fillWeatherForecast();
+  renderForecastWeather();
+};
+
+var fetchForecast = function (query) {
+  $.ajax({
+    method: "GET",
+    url: "https://api.openweathermap.org/data/2.5/forecast?q=" + query + "&units=imperial" + "&appid=" + apiKey,
+    dataType: "json",
+    success: function(data) {
+     addWeatherForecast(data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log(textStatus);
+    }
+  });
+};
+//makes an object with data organized by date. 
+var compileFetchData = function (objectArray, property) {
+  return objectArray.reduce(function (acc, objs) {
+    let key = objs[property].slice(0, 10)
+    if(!acc[key]) {
+      acc[key] = []
+    };
+    acc[key].push(objs)
+    return acc;
+  }, {});
+};
+
+//selects target data points and pushes data obj into weatherForecast array
+var buildWeatherForecastObjs = function (array) {
+  var temp = [];
+  var condition = [];
+  var icon = [];
+  var weekDays;
+
+  for (let i = 0; i < array.length; i++) {
+    array[i]['dt_txt'] = moment(array[i]['dt_txt']).format('dddd'); //weekday
+
+    temp.push(Math.round(array[i].main.temp|| null));
+    condition.push(array[i].weather[0].main || null);
+    icon.push(array[i].weather[0].icon || null);
+    weekDays = array[i].dt_txt || null;
+  };
+  
+  weatherForecast.push({
+    forecastWeatherCondition: condition[3],
+    forecastTemp: Math.max(...temp),
+    forecastWeatherIcon: "http://openweathermap.org/img/wn/" + icon[3] + "@2x.png" ,
+    forecastWeekDay: weekDays
+  })
+};
+
+var getToday = function () {
+  today = new Date();
+  var dd = String(today.getDate()).padStart(2, '0');
+  var mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0
+  var yyyy = today.getFullYear();
+  today = yyyy + '-' + mm + '-' + dd;
+  return today; 
+};
+
+//loops through dailyData and adds objs of target data to weatherForecast array via bulidWeatherForecastObjs
+//excludes current day's data
+var fillWeatherForecast = function () {
+  getToday();
```

Comment:
You should avoid doing this - instead, write something like:

```js
var today = getToday();
```

Again, you're creating a side effect b/c `getToday` mutating a global variable. You're right to return `today` from `getToday`, but you're not utilizing that returned value.